scrapinghub/client/utils.py

```diff
@@ -136,40 +137,31 @@ class _MappingProxy(_ProxyBase):
 
     def __init__(self, cls, client, key):
         super(_MappingProxy, self).__init__(cls, client, key)
-        methods = ['ignore_fields', 'expire', 'save', 'liveget']
-        self._proxy_methods(methods)
-        self.iterkeys = self.__iter__
-
-    def __str__(self):
-        return str(self._origin._data)
-
-    def __repr__(self):
-        return '{}({})'.format(self.__class__.__name__,
-                               repr(self._origin._data))
-
-    def __getitem__(self, key):
-        return self._origin[key]
-
-    def __setitem__(self, key, value):
-        self._origin[key] = value
 
-    def __delitem__(self, key):
-        del self._origin[key]
+        # cls_methods methods will be imported from origin class
+        origin_cls_methods = ['__iter__', '__len__', '__str__', '__repr__',
+                              '__getitem__', '__setitem__', '__delitem__']
 
-    def __iter__(self):
-        return iter(self._origin)
+        other_attrs = ['ignore_fields', 'expire', '_data', '_deleted',
+                       'save', 'liveget', ('iter', '__iter__')]
 
-    def __len__(self):
-        return len(self._origin)
+        # blacklisted attributes should be just skipped on proxying
+        blacklisted_attrs = ['__class__', '__init__', '__str__', '__repr__']
 
-    def keys(self):
-        return list(self.iterkeys())
+        # proxy all the main methods from MutableMapping
+        methods = [method for method in dir(MutableMapping)
+                   if callable(getattr(MutableMapping, method))
+                   and method not in origin_cls_methods
+                   and method not in blacklisted_attrs]
+        proxy_methods(MutableMapping, self.__class__, methods, force=True)
```

Comment:
re magic methods that already exist - if you follow example I provided you'll receive a list of methods to proxy which looks like this for Py2:

```
set(['__metaclass__', 'popitem', '__getattribute__', 'pop', '__sizeof__', '__setattr__', '__reduce_ex__', '__new__', '__contains__', '__format__', '_deleted', 'itervalues', 'save', '__len__', '__ne__', '__getitem__', 'get', 'keys', '__reduce__', ('iter', '__iter__'), 'update', 'liveget', '__iter__', '__subclasshook__', 'iteritems', '_data', '__eq__', 'iterkeys', '__delitem__', 'setdefault', 'items', 'clear', '__setitem__', '__delattr__', 'ignore_fields', 'expire', 'values'])
```

does this list include methods that should be proxied with a force option?